src/eu/janinko/games/backgammon/Board.java

```diff
@@ -1,195 +1,392 @@
-		default: return 0;
+		// Check the color and return its associated number of blotted stones
+		switch (color) {
+		case WHITE:
+			return barWhite;
+		case BLACK:
+			return barBlack;
+		default:
+			return 0;
 		}
 	}
 
-	public boolean canMove(int from, int count){
-		if(from < 0 || from > 24) return false;
-		
-		if(stoneCounts[from] == 0) return false;
+	/**
+	 * Checks if a specified move is leagal in the game of backgammon in term of
+	 * the board state
+	 * 
+	 * @param from
+	 *            - Position of a stone being moved
+	 * @param count
+	 *            - Number of positions being moved away from 'from'
+	 * @return If the Move is legal to move a piece in 'from' 'count' positions
+	 *         away
+	 */
+	public boolean canMove(int from, int count) {
+		// If from is not a bopard position the move is illegal
+		if (from < 0 || from > 24) {
+			return false;
+		}
+
+		// If there are no stones at from to move the move is illegal
+		if (stoneCounts[from] == 0) {
+			return false;
+		}
+		// Calculate the position of the target location
 		Stone.Color who = stoneColors[from];
 		int target;
-		if(who == Stone.Color.WHITE){
-			if(barWhite > 0) return false;
+		if (who == Stone.Color.WHITE) {
+			// If a blot is inplace no other moves may be done
+			if (barWhite > 0) {
+				return false;
+			}
+			// White moves in the positive direction
 			target = from + count;
-		}else{
-			if(barBlack > 0) return false;
+		} else {
+			// If a blot is inplace no other moves may be done
+			if (barBlack > 0) {
+				return false;
+			}
+			// Black moves in negative direction
 			target = from - count;
 		}
-		if(target > 23 || target < 0){
-			return hasAllInBase(who,from);
+
+		// If the request is a move off the board, only allow the move if all
+		// stones are in base
+		if (target > 23 || target < 0) {
+			return hasAllInBase(who, from);
 		}
+
+		// Check if the taget position can be moved to
 		Stone.Color tarwho = stoneColors[target];
-		if(tarwho == who || tarwho == Stone.Color.NONE){
+		if (tarwho == who || tarwho == Stone.Color.NONE) {
 			return true;
-		}else{
+		} else {
+			// Check if a blot is possible
 			return stoneCounts[target] == 1;
 		}
 	}
-	
-	public boolean canPut(Stone.Color color, int number){
-		switch(color){
+
+	/**
+	 * Checks if a stone can be put back into play from being blotted
+	 * 
+	 * @param color
+	 *            - Color of stone returning to play
+	 * @param number
+	 *            - Position in enemy home where the stone is to be put
+	 * @return If the a blotted stone of color 'color' can be be put in position
+	 *         'number'
+	 */
+	public boolean canPut(Stone.Color color, int number) {
+		switch (color) {
 		case WHITE:
-			if(barWhite == 0) return false;
-			if(stoneColors[number-1] == Stone.Color.BLACK) return false;
+			// If white has no blotted stones the play is illegal
+			if (barWhite == 0)
+				return false;
+			// If the opponent has a made then the play cannot be done
```

Comment:
A made in backgammon is when two stones reside on a point, such that an opponent cannot land on said point.